ticket_facts.view.lkml

```diff
@@ -0,0 +1,277 @@
+      sql: ${TABLE}.name ;;
+    }
+
+    dimension: via__source__to__name {
+      hidden: yes
+      type: string
+      sql: ${TABLE}.via__source__to__name ;;
+    }
+
+    dimension: multiple_groups_involved {
+      description: "Yes, if the ticket has changed assigned group at any point."
+      group_label: "Activity Details"
+      type: yesno
+      sql: ${TABLE}.number_of_groups_involved > 0 ;;
+    }
+
+    dimension: number_outbound_calls {
+      description: "Number of outbound calls associated with ticket"
+      group_label: "Activity Counts"
+      type: number
+      sql: ${TABLE}.number_outbound_calls ;;
+    }
+
+    dimension: has_outbound_calls {
+      description: "Yes, if ticket has associated outbound calls"
+      group_label: "Activity Details"
+      type: yesno
+      sql: ${number_outbound_calls} > 0 ;;
+    }
+
+    dimension: number_inbound_calls {
+      description: "Number of inbound calls associated with ticket"
+      group_label: "Activity Counts"
+      type: number
+      sql: ${TABLE}.number_inbound_calls ;;
+    }
+
+    dimension: has_inbound_calls {
+      description: "Yes, if ticket has associated inbound calls"
+      group_label: "Activity Details"
+      type: yesno
+      sql: ${number_inbound_calls} > 0 ;;
+    }
+
+    dimension: number_inbound_emails {
+      description: "Number of inbound emails associated with ticket"
+      group_label: "Activity Counts"
+      type: number
+      sql: ${TABLE}.number_inbound_emails ;;
+    }
+
+    dimension: has_inbound_emails {
+      description: "Yes, if ticket has associated inbound emails"
+      group_label: "Activity Details"
+      type: yesno
+      sql: ${number_inbound_emails} > 0 ;;
+    }
+
+    dimension: involves_customer {
+      description: "Yes, if ticket has inbound or outbound activity"
+      group_label: "Ticket Details"
+      type: yesno
+      sql: ${number_inbound_calls} > 0 OR ${number_inbound_emails} > 0 OR ${number_outbound_calls} > 0 OR ${number_outbound_emails} > 0 ;;
+    }
+
+    dimension: is_first_touch_resolution_email {
+      description: "Yes, if inbound email was solved at first agent touch"
+      group_label: "Activity Details"
+      type: yesno
+      sql: ${ticket_source} LIKE 'Inbound Email%'
+         AND ${status} IN ('solved', 'closed')
+         AND ${is_ticket_merged} = false
+         AND (${number_outbound_emails} + ${number_outbound_calls}) = 1
+         AND ${is_parent_to_merged_tickets} = false ;;
+    }
+
+    dimension: is_eligible_for_first_touch_resolution_email {
+      description: "Yes, if inbound email was solved and has outbound activity"
+      group_label: "Activity Details"
+      type: yesno
+      sql: ${ticket_source} LIKE 'Inbound Email%'
+          AND ${status} IN ('solved', 'closed')
+          AND ${is_ticket_merged} = false
+          AND (${number_outbound_emails} + ${number_outbound_calls}) >= 1 ;;
+    }
+
+    dimension: is_first_touch_resolution_phone_call {
```

Comment:
Why do you need this dimension? Couldn't you just have a combined `is_one_touch_resolved` and then allow the analyst to add a pivot by ${ticket_source} if they want to break it out by `email` vs `phone` vs `other`?

Reply:
It seems like the logic should be:

- "If the channel is asynchronous like email then we allow one reply before solve"
- "If the channel is asynchronous like phone then we do not allow any form of reply before solve"

For chat, we use "session" rather than "chat", so if there is an inbound chat, and we have one outbound chat session, then it remains "one touch". If we get an inbound chat but we don't respond (say it's after hours), and then have a single email reply, it's still "one touch".

Also, I think this logic is not correct for `Inbound Voicemail`. If we get a Voicemail, we may reply by email, outbound call, or by chat session. In either case, it should be "one touch" if the issue is resolved afterwards? In the logic below, an "Inbound Voicemail" will always have either an "Outbound Call" or an "Outbound Email" since we have to either call back or write back.

Reply:
this one might be helpful to review live tomorrow during our meeting.  Here is the section where this is discussed -- 

https://paper.dropbox.com/doc/Customer-Happiness-Dashboard-Review-921--ANQ_SDH1zJ66tkFZ~nLh9LOcAg-CofcJzFOC8w2D1wQKDm9z#:uid=047404549169033205488777&h2=One-touch-resolution

Because email and phone calls are so different in nature, I think it makes sense to separate them out and have different goals for both.  

Updated logic for voicemail.  I was originally thinking we wouldn't want to consider those one touch because we missed the opportunity to take care of it right away, but I see your point for sure.  Updated.

Reply:
done